src/Search/Variants/SearchVariant.php

```diff
@@ -166,6 +166,42 @@ public static function with($class = null, $includeSubclasses = true)
         return self::$call_instances[$key];
     }
 
+    /**
+     * Similar to {@link SearchVariant::with}, except will only use variants that apply to at least one of the classes
+     * in the input array, where {@link SearchVariant::with} will run the query on the specific class you give it.
+     *
+     * @param array $classes
+     * @return SearchVariant_Caller
+     */
+    public static function withCommon(array $classes = [])
+    {
+        // Allow caching
+        $cacheKey = serialize($classes);
```

Comment:
is this an array of strings? if so can we change the docblock to `string[]` please.

I'd also recommend `md5` the `cacheKey` as a string representing a serialized array seems excessive for an array key.

Reply:
Yep, fair call. I'll update